perceval/backends/core/rocketchat.py

```diff
@@ -0,0 +1,338 @@
+        """
+        result = json.loads(raw_room_info)
+        return result['channel']
+
+    @staticmethod
+    def parse_history(raw_history):
+        """Parse a room history JSON stream.
+        This method parses a JSON stream, containing the history of
+        a room, and returns a list with the parsed data.
+        :param raw_history: JSON string to parse
+        :returns: a tuple with a list of dicts with the parsed messages"""
+        result = json.loads(raw_history)
+        return result['messages']
+
+    @staticmethod
+    def parse_user(raw_user):
+        """Parse a user's info JSON stream.
+        This method parses a JSON stream, containing the information
+        from a user, and returns a dict with the parsed data.
+        :param raw_user: JSON string to parse
+        :returns: a dict with the parsed user's information
+        """
+        result = json.loads(raw_user)
+        return result['u']
+
+    @staticmethod
+    def metadata_category(item):
+        """Extracts the category from a Rocket chat item.
+        This backend only generates one type of item which is
+        'message'.
+        """
+        return CATEGORY_MESSAGE
+
+    @staticmethod
+    def metadata_id(item):
+        """Extracts the identifier from a RocketChat item."""
+        return item["_id"]
+
+    @staticmethod
+    def metadata_updated_on(item):
+        """Extracts the update time from a RocketChat item.
+        The timestamp is extracted from 'ts' field,
+        and sliced to get the year, month and date.
+        It is then converted into a UNIX timestamp.
+        :param item: item generated by the backend
+        :returns: extracted timestamp
+        """
+        ts = item['ts'][:10]
+        ts = time.mktime(datetime.datetime.strptime(ts, "%Y-%m-%d").timetuple())
+
+        return ts
+
+    def _init_client(self, from_archive=False):
+        """Init client"""
+
+        return RocketChatClient(self.url, self.user_id, self.api_token, self.max_items, self.archive,
+                                from_archive, self.ssl_verify, roomId=self.roomId)
+
+    def __get_or_fetch_user(self, user_id, username, name):
+        if user_id in self._users:
+            return self._users[user_id]
+
+        logger.debug("User %s not found on client cache; fetching it", user_id)
+
+        user = {
+            "username": username,
+            "name": name,
+        }
+
+        self._users[user_id] = user
+        return user
+
+
+class RocketChatClient(HttpClient):
+    """RocketChat API client.
+    Client for fetching information from the RocketChat server
+    using its REST API.
+    :param api_token: key needed to use the API
+    :param max_items: maximum number of items per request
+    :param archive: an archive to store/read fetched data
+    :param from_archive: it tells whether to write/read the archive
+    :param ssl_verify: enable/disable SSL verification
+    """
+    ROOM_HISTORY = 'channels.history'
```

Comment:
Following the pointer of @animeshk08 , I checked the two endpoints:
- https://rocket.chat/docs/developer-guides/rest-api/channels/history/
- https://rocket.chat/docs/developer-guides/rest-api/channels/messages/

From the description of the two endpoints I don't understand the difference. I see that the latter was added in a later release and has some additional filtering. Beyond that the data returned seems the same.

Is there a specific reason for selecting the `channels.history`?

Reply:
I had originally missed the query parameter while going through the documentation, which is why I stuck with channels.history. However, I spent a substantial amount of time the day yesterday trying to get date queries to work using either of the endpoints, and I came to the conclusion that date queries just don't work (at least I wasn't able to get them to work even spending a lot of time on them). 

For channels.history, I tried adding the oldest parameter, in various formats, including the one mentioned at [https://github.com/RocketChat/Rocket.Chat/issues/10871](https://github.com/RocketChat/Rocket.Chat/issues/10871), but the api sent the same response as the one without it specified. I then thought of moving to channels.messages and went over the documentation for the query parameter to try and make channels.messages work, but I ran into a `invalid-query` error for queries that supposedly worked for other people, and was stuck on it before realizing my mistake. However, even after getting it to work and spending quite some time on getting it to work, I still kept getting the following response:

    {
        "messages": [],
        "success": true
    }

By this point, I was convinced that date queries don't work, but in a last effort, tried adding the `latest` parameter to the channels.history endpoint, and this time got the same result as above.

This is why I am pretty convinced that date queries don't work with the rocket chat api.

Reply:
Ok, thank you @Syn3rman for reporting this useful information. Once the PR is ready for review, I'll have a look and evaluate possible workarounds (if exist).